components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConfigurationServiceImpl.java

```diff
@@ -166,4 +170,55 @@ public void triggerRemoteFetch(RemoteFetchConfiguration fetchConfiguration) thro
                     fetchConfiguration.getRemoteFetchConfigurationId());
         }
     }
+
+    /**
+     * Validate limit.
+     * Check optionalLimit has a value, or else set to default value.
+     * @param optionalLimit given limit value.
+     * @return validated limit and offset value.
+     */
+    private int validateLimit(OptionalInt optionalLimit) throws RemoteFetchCoreException {
+
+        if (!(optionalLimit.isPresent())) {
+            if (log.isDebugEnabled()) {
+                log.debug("Given limit is null. Therefore we get the default limit from " +
+                        "identity.xml.");
+            }
+        }
+        int limit = optionalLimit.orElse(RemoteFetchConfigurationUtils.getDefaultItemsPerPage());
+
+        if (limit < 0) {
+            String message = "Given limit: " + limit + " is a negative value.";
+            throw new RemoteFetchCoreException("Unable to retrieve remote fetch configuration list " +
+                    message);
+        }
+
+        int maximumItemsPerPage = RemoteFetchConfigurationUtils.getMaximumItemPerPage();
+        if (limit > maximumItemsPerPage) {
+            if (log.isDebugEnabled()) {
+                log.debug("Given limit exceed the maximum limit. Therefore we get the default limit from " +
```

Comment:
log.debug("Given limit exceed the maximum limit. Therefore setting the max limit from " +
                        "identity.xml. limit